app/src/main/java/ch/hikemate/app/ui/profile/ProfileScreen.kt

```diff
@@ -0,0 +1,149 @@
+                    .padding(p)
+                    .padding(
+                        // Add padding to the sidebar padding
+                        start = 16.dp,
+                        end = 16.dp,
+                        top = 16.dp,
+                    ),
+            verticalArrangement = Arrangement.spacedBy(16.dp)) {
+              Text(
+                  context.getString(R.string.profile_screen_title),
+                  style = TextStyle(fontWeight = FontWeight.Bold, fontSize = 32.sp),
+                  modifier = Modifier.testTag(ProfileScreen.TEST_TAG_TITLE))
+
+              DisplayInfo(
+                  context.getString(R.string.profile_screen_name_label),
+                  profile.name,
+                  Modifier.testTag(ProfileScreen.TEST_TAG_NAME))
+              DisplayInfo(
+                  context.getString(R.string.profile_screen_email_label),
+                  profile.email,
+                  Modifier.testTag(ProfileScreen.TEST_TAG_EMAIL))
+              DisplayInfo(
+                  context.getString(R.string.profile_screen_hiking_level_label),
+                  buildString {
+                    append(context.getString(R.string.profile_screen_hiking_level_info_prefix))
+                    append(" ")
+                    append(
+                        when (profile.hikingLevel) {
+                          HikingLevel.BEGINNER ->
+                              context.getString(R.string.profile_screen_hiking_level_info_beginner)
+                          HikingLevel.INTERMEDIATE ->
+                              context.getString(
+                                  R.string.profile_screen_hiking_level_info_intermediate)
+                          HikingLevel.EXPERT ->
+                              context.getString(R.string.profile_screen_hiking_level_info_expert)
+                        })
+                    append(" ")
+                    append(context.getString(R.string.profile_screen_hiking_level_info_suffix))
+                  },
+                  Modifier.testTag(ProfileScreen.TEST_TAG_HIKING_LEVEL))
+              DisplayInfo(
+                  context.getString(R.string.profile_screen_join_date_label),
+                  DateFormat.getDateInstance(DateFormat.LONG).format(profile.joinedDate.toDate()),
+                  Modifier.testTag(ProfileScreen.TEST_TAG_JOIN_DATE))
+
+              BigButton(
+                  buttonType = ButtonType.PRIMARY,
+                  label = context.getString(R.string.profile_screen_edit_profile_button_text),
+                  onClick = { navigationActions.navigateTo(Screen.EDIT_PROFILE) },
+                  Modifier.testTag(ProfileScreen.TEST_TAG_EDIT_PROFILE_BUTTON))
+
+              BigButton(
+                  buttonType = ButtonType.SECONDARY,
+                  label = context.getString(R.string.profile_screen_sign_out_button_text),
+                  onClick = {
+                    // TODO: Sign out
```

Comment:
Question: Same as before, why not implementing this code with a dummy repository? Looks better to me

Reply:
Done!